CHANGELOG.md

```diff
@@ -1,5 +1,441 @@
+
+See also this [testing of profile migrations](https://github.com/aiidateam/aiida-core/discussions/5379), for some indicative timings.
+:::
+
+### Improvements to the AiiDA ORM 👌
+
+#### Node repository
+
+Inline with the storage improvements, {class}`~aiida.orm.Node` methods associated with the repository have some backwards incompatible changes:
+
+:::{dropdown} `Node` repository method changes
+
+Altered:
+
+- `FileType`: moved from `aiida.orm.utils.repository` to `aiida.repository.common`
+- `File`: moved from `aiida.orm.utils.repository` to `aiida.repository.common`
+- `File`: changed from namedtuple to class
+- `File`: can no longer be iterated over
+- `File`: `type` attribute was renamed to `file_type`
+- `Node.put_object_from_tree`: `path` argument was renamed to `filepath`
+- `Node.put_object_from_file`: `path` argument was renamed to `filepath`
+- `Node.put_object_from_tree`: `key` argument was renamed to `path`
+- `Node.put_object_from_file`: `key` argument was renamed to `path`
+- `Node.put_object_from_filelike`: `key` argument was renamed to `path`
+- `Node.get_object`: `key` argument was renamed to `path`
+- `Node.get_object_content`: `key` argument was renamed to `path`
+- `Node.open`: `key` argument was renamed to `path`
+- `Node.list_objects`: `key` argument was renamed to `path`
+- `Node.list_object_names`: `key` argument was renamed to `path`
+- `SinglefileData.open`: `key` argument was renamed to `path`
+- `Node.open`: can no longer be called without context manager
+- `Node.open`: only mode `r` and `rb` are supported, [use `put_object_from_` methods instead](https://github.com/aiidateam/aiida-core/issues/4721#issuecomment-920100415)
+- `Node.get_object_content`: only mode `r` and `rb` are supported
+- `Node.put_object_from_tree`: argument `contents_only` was removed
+- `Node.put_object_from_tree`: argument `force` was removed
+- `Node.put_object_from_file`: argument `force` was removed
+- `Node.put_object_from_filelike`: argument `force` was removed
+- `Node.delete_object`: argument `force` was removed
+
+Added:
+
+- `Node.walk`
+- `Node.copy_tree`
+- `Node.is_valid_cache` setter
+- `Node.objects.iter_repo_keys`
+
+Additionally, `Node.open` should always be used as a context manager, for example:
+
+```python
+with node.open('filename.txt') as handle:
+    content = handle.read()
+```
+
+:::
+
+#### QueryBuilder
+
+When using the {class}`~aiida.orm.QueryBuilder` to query the database, the following changes have been made:
+
+- The `Computer`'s `name` field is now replaced with `label` (as previously deprecated in v1.6)
+- The `QueryBuilder.queryhelp` attribute is deprecated, for the `as_dict` (and `from_dict`) methods
+- The `QueryBuilder.first` method now allows the `flat` argument, which will return a single item, instead of a list of one item
```

Comment:
```suggestion
- The `QueryBuilder.first` method now allows the `flat` argument, which will return a single item, instead of a list of one item, if only a single projection is defined.
```